File: application/credit-card-enrollment-backend-in-java/src/main/java/cloud/ambar/common/ambar/httpresponse/AmbarResponseFactory.java
package cloud.ambar.common.ambar.httpresponse;

import java.util.Arrays;

public class AmbarResponseFactory {
    public static String retryResponse(Exception exception) {
        return "{\"result\":{\"error\":{\"policy\":\"MUST_RETRY\",\"class\":\"" + exception.getClass() + "\",\"description\":\"message:" + exception.getMessage() + "\"}}}";
    }

    public static String successResponse() {
        return "{\"result\":{\"success\":{}}}";
    }
}

File: application/backend-credit-card-enrollment/backend-typescript/src/common/ambar/AmbarResponseFactory.ts
export class AmbarResponseFactory {
    static retryResponse(exception: Error): string {
        const message = exception.message.replace(/"/g, '\\"');
        return `{"result":{"error":{"policy":"must_retry","class":"${exception.constructor.name}","description":"message:${message}"}}}`;
    }

    static successResponse(): string {
        return '{"result":{"success":{}}}';
    }
}